core/server/master/src/main/java/alluxio/master/metrics/MetricsStore.java

```diff
@@ -73,12 +75,15 @@ private static String getFullInstanceId(String hostname, String id) {
     return str;
   }
 
+  // The lock for worker and client metrics set
```

Comment:
Can you add a bit more details in this comment? I don't think it is obvious how to use the read and write lock, since the "put*" methods just take a read lock.

Reply:
Added more details! Thanks for pointing out